src/pymodaq_plugins_keithley/daq_viewer_plugins/plugins_0D/daq_0Dviewer_Keithley2750.py

```diff
@@ -0,0 +1,89 @@
+from pymodaq.control_modules.viewer_utility_classes import comon_parameters
+from pymodaq_plugins_keithley import config
+from pymodaq_plugins_keithley.hardware.keithley27XX.Keithley27XX_Viewer import DAQ_0DViewer_Keithley27XX
+from pymodaq_plugins_keithley.hardware.keithley27XX.keithley2750_VISADriver import Keithley2750VISADriver as Keithley
+from pymodaq.utils.logger import set_logger, get_module_name
+
+logger = set_logger(get_module_name(__file__))
+
+
+class DAQ_0DViewer_Keithley2750(DAQ_0DViewer_Keithley27XX):
+    """ Keithley viewer class for Keithley 2750 Multimeter/Switch System.
+    """
+    model = "2750"
+    resources_list = []
+
+    # Read configuration file
+    for instr in config["Keithley", model].keys():
+        if "INSTRUMENT" in instr:
+            resources_list += [config["Keithley", model, instr, "rsrc_name"]]
+    logger.info("resources list = {}".format(resources_list))
+
+    params = comon_parameters + [
+        {'title': 'Resources', 'name': 'resources', 'type': 'list', 'limits': resources_list,
+         'value': resources_list[0]},
+        DAQ_0DViewer_Keithley27XX.params
+    ]
+
+    def __init__(self, parent=None, params_state=None):
+        super().__init__(parent, params_state)
+
+    def ini_detector(self, controller=None):
+        """Detector communication initialization
+
+        :param controller: Custom object of a PyMoDAQ plugin (Slave case). None if one actuator/detector by controller.
+        :type controller: object
+
+        :return: Initialization status, false if it failed otherwise True
+        :rtype: bool
+        """
+        logger.info("Detector 0D initialized")
+
+        if self.settings.child('controller_status').value() == "Slave":
+            if controller is None:
+                raise Exception('no controller has been defined externally while this detector is a slave one')
+            else:
+                self.controller = controller
+        else:
+            try:
+                # Select the resource to connect with and load the dedicated configuration
```

Comment:
couldn't you do in the base class a method to encapsulate all this and refactor even more the code? I feel that most of the code here is still identical